src/components/settings-address-book-edit.tsx

```diff
@@ -0,0 +1,145 @@
+import React, {memo, useState} from 'react';
+
+import {Alert, StyleSheet, View} from 'react-native';
+
+import {
+  Button,
+  ButtonVariant,
+  Icon,
+  IconButton,
+  Input,
+  KeyboardSafeArea,
+} from './ui';
+
+import {useContacts} from '../contexts/contacts';
+import {useTypedNavigation} from '../hooks';
+import {I18N, getText} from '../i18n';
+import {LIGHT_BG_7, LIGHT_GRAPHIC_BASE_2} from '../variables';
+
+interface SettingsAddressBookEditProps {
+  initName?: string;
+  initAddress?: string;
+  buttonType?: 'save' | 'del';
+  isCreate?: boolean;
+  isEdit?: boolean;
+  onChangeAddress?: (text: string) => void;
+}
+
+export const SettingsAddressBookEdit = memo(
+  ({
+    initAddress = '',
+    initName = '',
+    buttonType = 'save',
+    isCreate = false,
+    isEdit,
+    onChangeAddress,
+  }: SettingsAddressBookEditProps) => {
+    const {goBack} = useTypedNavigation();
+
+    const contacts = useContacts();
+    const [inputName, setInputName] = useState(initName);
+
+    const onSubmit = () => {
+      if (isCreate) {
+        contacts.createContact(initAddress, inputName);
+      } else {
+        contacts.updateContact(initAddress, inputName);
+      }
```

Comment:
these logic can be moved on top like

`   const onSubmit = () => {
  onSubmitFromProps(inputName)
       goBack();
   };
`

and in screen 

`
const onSubmit = (inputName) => {
     if (isCreate) {
         contacts.createContact(initAddress, inputName);
       } else {
         contacts.updateContact(initAddress, inputName);
       }
}
`